common/persistence/sql/sql_execution_store_test.go

```diff
@@ -1663,3 +1663,223 @@ func TestRangeCompleteTimerTask(t *testing.T) {
+					TaskType:                1,
+					Version:                 202,
+					FirstEventID:            10,
+					NextEventID:             101,
+					ScheduledID:             19,
+					EventStoreVersion:       persistence.EventStoreVersion,
+					NewRunEventStoreVersion: persistence.EventStoreVersion,
+					BranchToken:             []byte(`bt`),
+					NewRunBranchToken:       []byte(`nbt`),
+					CreationTimestamp:       time.Unix(1, 1),
+				}).Return(persistence.DataBlob{Data: []byte(`replication`), Encoding: "replication"}, nil)
+				mockDB.EXPECT().InsertIntoReplicationTasksDLQ(gomock.Any(), &sqlplugin.ReplicationTaskDLQRow{
+					SourceClusterName: "source",
+					ShardID:           shardID,
+					TaskID:            101,
+					Data:              []byte(`replication`),
+					DataEncoding:      "replication",
+				}).Return(nil, nil)
+			},
+			wantErr: false,
+		},
+		{
+			name: "Error case - failed to encode data",
+			req: &persistence.InternalPutReplicationTaskToDLQRequest{
+				SourceClusterName: "source",
+				TaskInfo: &persistence.InternalReplicationTaskInfo{
+					DomainID:          "abdcea69-61d5-44c3-9d55-afe23505a542",
+					WorkflowID:        "test",
+					RunID:             "abdcea69-61d5-44c3-9d55-afe23505a54a",
+					TaskType:          1,
+					TaskID:            101,
+					Version:           202,
+					FirstEventID:      10,
+					NextEventID:       101,
+					ScheduledID:       19,
+					BranchToken:       []byte(`bt`),
+					NewRunBranchToken: []byte(`nbt`),
+					CreationTime:      time.Unix(1, 1),
+				},
+			},
+			mockSetup: func(mockDB *sqlplugin.MockDB, mockParser *serialization.MockParser) {
+				mockParser.EXPECT().ReplicationTaskInfoToBlob(gomock.Any()).Return(persistence.DataBlob{}, errors.New("some error"))
+			},
+			wantErr: true,
+		},
+		{
+			name: "Error case - failed to insert into database",
+			req: &persistence.InternalPutReplicationTaskToDLQRequest{
+				SourceClusterName: "source",
+				TaskInfo: &persistence.InternalReplicationTaskInfo{
+					DomainID:          "abdcea69-61d5-44c3-9d55-afe23505a542",
+					WorkflowID:        "test",
+					RunID:             "abdcea69-61d5-44c3-9d55-afe23505a54a",
+					TaskType:          1,
+					TaskID:            101,
+					Version:           202,
+					FirstEventID:      10,
+					NextEventID:       101,
+					ScheduledID:       19,
+					BranchToken:       []byte(`bt`),
+					NewRunBranchToken: []byte(`nbt`),
+					CreationTime:      time.Unix(1, 1),
+				},
+			},
+			mockSetup: func(mockDB *sqlplugin.MockDB, mockParser *serialization.MockParser) {
+				mockParser.EXPECT().ReplicationTaskInfoToBlob(gomock.Any()).Return(persistence.DataBlob{Data: []byte(`replication`), Encoding: "replication"}, nil)
+				err := errors.New("some error")
+				mockDB.EXPECT().InsertIntoReplicationTasksDLQ(gomock.Any(), gomock.Any()).Return(nil, err)
+				mockDB.EXPECT().IsDupEntryError(err).Return(false)
+				mockDB.EXPECT().IsNotFoundError(err).Return(true)
```

Comment:
nit: I prefer to use `Times(n)` if it's predictable. just provides more strict validation.